File: app/src/main/java/com/organizer/DateController.java
package com.organizer;

import java.util.HashMap;
import java.util.List;

public class DateController
{
    public HashMap<Integer, List<Task>> dictionary = new HashMap<>();
    public HashMap<Integer, Day> days = new HashMap<>();
    int todayD = 7;
    int todayM = 4;
    int todayY = 2021;
    public int today = buildId(todayD, todayM, todayY);
    public int textViewSize;
    
    public enum Weekdays
    {
        SU,
        MO,
        TU,
        WE,
        TH,
        FR,
        SA
    }
    
    public void addDays(int numDays, int offset)
    {
        for (int i = offset; i < numDays + offset; i++)
        {
            days.put(i, new Day(buildId(todayD + i, todayM, todayY))); // TODO offset rip
        }
    }
    
    public int buildId(int d, int m, int y)
    {
        return (y * 10000) + (m * 100) + d;
    }
    
    public int buildD(int id)
    {
        return id - (buildM(id) * 100) - (buildY(id) * 10000);
    }
    
    public int buildM(int id)
    {
        return id / 100 - (buildY(id) * 100);
    }
    
    public int buildY(int id)
    {
        return id / 10000;
    }
    
    public boolean idExists(int id)
    {
        Day day = new Day(id);
        return day.dayExists();
    }
    
    public void idDebug(int id)
    {
        Day day = new Day(id);
        day.dayDebug();
    }
    
    int calculateHeight(double start, double end)
    {
        double dayStart = 6;
        double dayEnd = 24;
        double contentHeight = MainActivity.getDisplayMetricsController().getScreenHeight() - textViewSize;
        return (int) (contentHeight * ((end - start) / (dayEnd - dayStart)));
    }
    
    int calculateY(double start)
    {
        double dayStart = 6;
        double dayEnd = 24;
        double contentHeight = MainActivity.getDisplayMetricsController().getScreenHeight() - textViewSize;
        return (int) (textViewSize + contentHeight * ((start - dayStart) / (dayEnd - dayStart)));
    }
}

File: app/src/main/kotlin/com/organizer/DateController.kt
package com.organizer

import com.organizer.MainActivity.Companion.inject

@Suppress("unused")
class DateController {

    private val displayController: DisplayMetricsController by inject()

    var dictionary = HashMap<Int, MutableList<CalendarEvent>>()
    var days = HashMap<Int, Day>()
    var todayD = 7
    var todayM = 4
    var todayY = 2021
    var today = buildId(todayD, todayM, todayY)
    var textViewSize = 0

    enum class Weekdays {
        SU, MO, TU, WE, TH, FR, SA
    }

    fun addDays(numDays: Int, offset: Int) {
        for (i in offset until numDays + offset) {
            days[i] = Day(buildId(todayD + i, todayM, todayY)) // TODO offset rip
        }
    }

    fun buildId(d: Int, m: Int, y: Int): Int {
        return y * 10000 + m * 100 + d
    }

    fun buildD(id: Int): Int {
        return id - buildM(id) * 100 - buildY(id) * 10000
    }

    fun buildM(id: Int): Int {
        return id / 100 - buildY(id) * 100
    }

    fun buildY(id: Int): Int {
        return id / 10000
    }

    fun idExists(id: Int): Boolean {
        val day = Day(id)
        return day.dayExists()
    }

    fun idDebug(id: Int) {
        val day = Day(id)
        day.dayDebug()
    }

    fun calculateHeight(start: Double, end: Double): Int {
        val dayStart = 6.0
        val dayEnd = 24.0
        val contentHeight = (displayController.screenHeight - textViewSize).toDouble()
        return (contentHeight * ((end - start) / (dayEnd - dayStart))).toInt()
    }

    fun calculateY(start: Double): Int {
        val dayStart = 6.0
        val dayEnd = 24.0
        val contentHeight = (displayController.screenHeight - textViewSize).toDouble()
        return (textViewSize + contentHeight * ((start - dayStart) / (dayEnd - dayStart))).toInt()
    }
}
